File: laba4/laba4/Line.cpp
/*Реалізувати клас Line. Лінійне рівняння y = Ax + B. Поле first – дійсне число,
коефіцієнт А; поле second – дійсне число, коефіцієнт В(A не дорівнює нулю).

Реалізувати метод root() – обчислення кореня лінійного рівняння.Метод має
перевіряти нерівність коефіцієнта A нулю.
*/
#include "Line.h"

Line::Line() {
    this->first = 0;
    this->second = 0;
}

Line::Line(int first, int second) {
    this->first = first;
    this->second = second;
}

Line::Line(Line& l) {
    this->first = l.first;
    this->second = l.second;
}

Line::~Line() {

}

bool Line::Init(int A, int B)
{
    if (A != 0)
    {
        return true;
    }
    else
    {
        return false;
    }
}
void Line::Read()
{
    int A, B;
    do
    {
        cout << "Please, throw me some numbers:" << endl;
        cout << " A = "; cin >> A;
        cout << " B = "; cin >> B;
    } while (!Init(A, B));
    Line temp(A, B);
    this->first = temp.first;
    this->second = temp.second;
}
void Line::Display() const
{
    cout << endl;
    cout << "----------------------------------------------------------" << endl;
    cout << " A = " << first << endl;
    cout << " B = " << second << endl;
    cout << "----------------------------------------------------------" << endl;
    cout << endl;
}


double Line::root()
{
    return second * (-1) / first;
}

File: l_1_C#/l_1_/Line.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace l_1_
{
    internal class Line
    {
        private double first;
        private double second;

        public double GetFirst()
        {
            return first;
        }

        public double GetSecond()
        {
            return second;
        }

        public void SetFirst(int first)
        {
            if (first != 0)
            {
                this.first = first;
            }
        }

        public void SetSecond(int second)
        {
            this.second = second;
        }

        public bool Init(double a, double b)
        {
            if (a != 0)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
        public void Read()
        {
            double A = 0, B = 0;
            do
            {
                System.Console.WriteLine("first = ");
                A = Convert.ToInt32(Console.ReadLine());
                System.Console.WriteLine("second = ");
                B = Convert.ToInt32(Console.ReadLine());
            } while (!Init(A, B));
            first = A;
            second = B;
        }

        public void Display()
        {
            Console.WriteLine("first: " + first);
            Console.WriteLine("second: " + second);
        }


        public double Function()
        {
            return second * (-1) / first;
        }
}
}
